Chapter Table UI Improvements

purpxd opened the pull request:
### **Chapter Table Checkbox Column**
**Before:** Checkbox selects all rows in the table
**After:** Checkbox selects all rows in current page
I figured maybe it wasn't intended to do that since there was pagination for the tables that the checkbox should only select only the chapters in the current page. Anyway if it was then I added the select all in the options dropdown.

### **Options Dropdown**
Moved the download and marking operations into the options dropdown. I think you wanted some confirmation for the downloads, this solves that. The mark selected operations I moved into the dropdown might be a little redundant since I kept the previous buttons where they were. But included mark all chapters which can be useful

### **Filter Dropdown**
The buttons row was getting too packed and with the addition of the read chapters filter I figured it would be appropriate to move it to a dropdown

### **Hide Read Filter**
This filter solves #326  


![image](https://github.com/user-attachments/assets/17775b42-37a2-4e7f-a48c-4548602f505e)

purpxd committed:
fix chapter table checkbox
Changes the column checkbox to select rows in the current page rather
than all rows. This will reduce clicks from users when marking large
series that aren't fully completed.

purpxd committed:
add options nested dropdown
*Moved marked buttons into dropdown, seperating operations for all rows
and selected rows
*Moved Download chapters into dropdown, also seperated for all and
selected
*Moved back the selected buttons back into the main buttons rows using outline variant

purpxd committed:
added filter dropdown
*consolidated group and language filters under the filter dropdown

purpxd committed:
modify popover filters
*Removed filter logo
*Swap popover trigger style from button to div
*Have popover show to the right
*Conditionally render chevron

purpxd committed:
added chapter table read filter
*Added read filter component to filter dropdown
*Modified library state to filter read chapters

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTable.tsx:
```diff
@@ -311,75 +333,197 @@ export function ChapterTable(props: ChapterTableProps) {
+                          <span>Read</span>
+                        </div>
+                      </DropdownMenuItem>
+                      <DropdownMenuSeparator />
```
remove this separator

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTable.tsx:
```diff
@@ -311,75 +333,197 @@ export function ChapterTable(props: ChapterTableProps) {
+                          <span>All Chapters</span>
+                        </div>
+                      </DropdownMenuItem>
+                      <DropdownMenuSeparator />
```
remove this separator

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTableReadFilter.tsx:
```diff
@@ -0,0 +1,27 @@
+export function ChapterTableReadFilter() {
+  const [hideReadChapters, setHideReadChapters] = useRecoilState(ReadChaptersState);
+
+  const toggleReadChapters = () => {
+    setHideReadChapters(!hideReadChapters);
+  };
+
+  return (
+    <div
+      className="flex items-center space-x-2 cursor-pointer"
+      onClick={toggleReadChapters}
+    >
+      {hideReadChapters ? (
+        <Check className="h-5 w-5 text-blue-500" />
+      ) : (
+        <Square className="h-5 w-5 text-gray-500" />
+      )}
+      <span className="text-sm">
+        Hide Read
+      </span>
+    </div>
+  );
```
- change to "Hide read"
- I think the check option is better as an actual checkbox, although it's fine to have the click event on the container div. Also would prefer to not have the menu close when toggling this option

```ts
return (
    <div
      className="flex items-center space-x-2 cursor-pointer"
      onClick={(e) => {
        e.stopPropagation();
        setHideReadChapters(!hideReadChapters);
      }}
    >
      <Checkbox checked={hideReadChapters} />
      <span className="text-sm">Hide read</span>
    </div>
  );
```

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTable.tsx:
```diff
@@ -311,75 +333,197 @@ export function ChapterTable(props: ChapterTableProps) {
+        <div className="flex space-x-2">
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button variant="outline">
+                <Filter className="w-4 h-4" />
+                Filters
+              </Button>
```
doesn't need to be part of this PR, but I think it would be good to have a badge with `X selected` on the button when either language/group filters are applied.

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTable.tsx:
```diff
@@ -269,6 +275,11 @@ export function ChapterTable(props: ChapterTableProps) {
   };
 
+  const getAllChapters = (): Chapter[] => {
+    console.log(table.getCoreRowModel())
```
remove this

xgi commented on apps/desktop/src/renderer/components/library/series/chapter-table/ChapterTable.tsx:
```diff
@@ -311,75 +333,197 @@ export function ChapterTable(props: ChapterTableProps) {
+              </Button>
             </div>
-          </>
-        )}
```
You can just get rid of these buttons

xgi commented on apps/desktop/src/renderer/state/libraryStates.ts:
```diff
@@ -103,8 +108,8 @@ export const sortedFilteredChapterListState = selector<Chapter[]>({
     const chapterListChOrder = get(chapterListChOrderState);
-
     const uniqueChapters = new Map();
+    const toggleReadChapters = get(ReadChaptersState);
```
`toggleReadChapters` doesn't make sense, this is a boolean. Maybe also change `ReadChaptersState` to something like `hideUnreadChaptersState` and adjust accordingly?

xgi reviewed (commented):
A few minor comments but this looks great. Thanks!

purpxd committed:
add filters badge indicator
*filters badge indicator that will also clear filters
*change hide read filter to use actual checkbox
*clean up

purpxd commented:
Thanks for the feedback xgi. This commit should do it, cheers!

purpxd committed:
add scroll to chapter table body
*removed overflow class in table div, it prevented sticky table headers
